pydis_core/_bot.py

```diff
@@ -231,6 +231,12 @@ async def wait_until_guild_available(self) -> None:
         """
         await self._guild_available.wait()
 
+    async def process_commands(self, message: discord.Message) -> None:
+        """Wait until all extensions are loaded before processing commands."""
```

Comment:
The docstring isn't quite true here as that's not the purpose of the async method.

Reply:
What do you mean, this is exactly what it's doing? It waits for the extension task to be complete (all extensions are laoded) and then processes commands.

Reply:
Processing commands is what it's intended to do, so that's what should go in the summary line.

The checks inside it are "business rules" / custom behavior we add to it, which is an implementation detail. 

So you can make it a multineline docstring and in the description, you say that it runs some validation, e.g. waits until extensions are loaded.

Reply:
https://github.com/python-discord/bot-core/pull/190/commits/2da733d7febe4bc0ff78fd0074811b829118161c